gradle-plugin/build.gradle

```diff
@@ -62,12 +62,11 @@ ktlint {
 }
 
 dependencies {
-  implementation libs.kotlin.gradlePlugin
-  implementation libs.kotlin.gradlePluginApi
-
   // Necessary to bump a transitive dependency.
   compileOnly libs.kotlin.reflect
 
-  // Compile only so that Java / Kotlin modules don't download it unnecessarily.
+  // Compile only so we don't pre-empty what the consuming project actually uses
+  compileOnly libs.kotlin.gradlePlugin
+  compileOnly libs.kotlin.gradlePluginApi
```

Comment:
Would this be a visible change to consumers / worth calling out in the changelog?

On that note, can you update the changelog for the overall PR? Going back through all the commits later would be a bit of a hassle unless you want to rebase and squash some of them 😁

Reply:
Definitely good to mention in the changelog. Will add it 👍

Reply:
Done